contributors/devel/release/scalability-validation.md

```diff
@@ -0,0 +1,123 @@
+- Make scalability validation a release prerequisite
+- Clearly lay down responsibilities for scale tests
+
+## Non-Goals
+
+We do not intend to:
+
+- Define the set of tests that comprise scalability and correctness suite
+- Define SLIs/SLOs (that’s discussed [here]) and thresholds for the tests
+- Discuss particular performance issues we’re facing wrt different releases
+
+While these are interesting from scalability perspective, they are not testing process issues but details of the tests themselves. And can change without requiring much changes in the process.
+
+## Proposed design
+
+For each of the problems above we propose solutions and discuss some caveats. This is an open discussion and better solutions may evolve in future.
+
+### Automate large cluster tests
+
+Currently, there are 2 kinds of tests needed to validate the claim “kubernetes supports X-node clusters”. These are:
+
+- Correctness tests - e2e tests verifying expected system behaviours
+- Performance tests - e2e tests for stress-testing the system perf ([Feature:Performance])
+
+We need to run them on 5k-node clusters, but they’re:
+
+- Time-consuming (12-24 hrs)
+- Expensive (tens of thousands of core hours per run)
+- Blocking other large tests (quota limitations + only one large test project available viz. 'kubernetes-scale')
+
+So we don’t want to run them too frequently. On the other hand, running them too infrequently means late identification and piling up of regressions. So we choose the following middleground:  
+(B = release-blocking, NB = not release-blocking)
+
+- Performance tests on 2k-node cluster in GCE/GKE alternatingly each week (NB)
+  - would catch most scale-related regressions
+  - would catch bugs in both GCE and GKE
+- Performance tests on 5k-node cluster in GCE each week (B)
+  - would catch scale-related bugs missed by 2k-node test (should be rare)
+- Correctness tests on 2k-node cluster in GCE/GKE alternatingly each week (NB)
+  - correctness tests failing on just large clusters is rare, so weekly is enough
+  - would catch bugs in both GCE and GKE
+- Correctness tests on 5k-node cluster in GCE/GKE alternatingly each week (B for GCE)
+  - would catch bugs left out by 2k-node (should be rare)
+  - would verify 5k-node clusters on GKE but without blocking the release
+
+Here's the proposed schedule (may be fine-tuned later based on test health / release schedule):
+
+| | Mon | Tue | Wed | Thu | Fri | Sat | Sun
+| ---- | :----: | :----: | :----:| :----: | :----: | :----: | :----:
+| Week 2k | GCE 5k perf (B) | | | GCE 5k corr (B) | | GKE 2k perf+corr (NB) |
+| Week 2k+1 | GCE 5k perf (B) | | | GKE 5k corr (NB) | | GCE 2k perf+corr (NB) |
+
+Why this schedule?
+
+- 5k tests might need special attention in case of failures so they should run on weekdays
+- Running 5k perf test on Mon gives the whole week for debugging in case of failures
+- Running 5k corr test on Thu gives Tue-Wed to debug any setup issues found on Mon
```

Comment:
Makes sense. It's a trade-off between having more days to fix 5k-perf bugs vs. freeing up a weekday by running 2k-node on weekend. We can tune this later as apt.